nmigen_boards/basys3.py

```diff
@@ -0,0 +1,115 @@
+import textwrap
+import subprocess
+
+from nmigen.vendor.xilinx_7series import *
+from nmigen_boards.resources import *
+from nmigen.build import *
+
+__all__ = ["Basys3Platform"]
+
+class Basys3Platform(Xilinx7SeriesPlatform):
+    """
+    Platform module for Digilent Basys 3
+
+    Based on:
+    https://reference.digilentinc.com/reference/programmable-logic/basys-3/
+    https://github.com/Digilent/digilent-xdc/blob/master/Basys-3-Master.xdc
+    """
+
+    device = "xc7a35t"
+    package = "cpg236"
+    speed = "1"
+    default_clk = "clk100"
+
+    _basys_iostandard = "LVCMOS33"
+    _default_attrs = Attrs(IOSTANDARD=_basys_iostandard)
```

Comment:
Please inline these. They do not change once correct because the board is fixed, so factoring them out is not useful, and it is easier to read the code where the attributes are present where they are used.

Reply:
fixed